src/sdk/pynni/nni/compression/torch/speedup/infer_shape.py

```diff
@@ -725,21 +960,19 @@ def conv2d_outshape(module_masks, mask):
     assert isinstance(mask, CoarseMask)
     assert mask.mask_index[1] is not None
     assert mask.mask_index[0] is None
-    assert mask.mask_index[2] is None
-    assert mask.mask_index[3] is None
 
-    if module_masks.output_mask is not None:
-        assert isinstance(module_masks.output_mask, CoarseMask)
-        # set shape of output
-        mask = module_masks.output_mask.merge(mask)
+    if module_masks.output_mask is None:
+        module_masks.output_mask = mask
     else:
+        # mask conflict should be solved by fix_mask_conflict before speedup
+        # mask and module_masks.output_mask may have different number of dimensions
+        # since they could be passed by linear or conv2d
+        assert all(module_masks.output_mask.mask_index[1] == mask.mask_index[1])
+
+    # shape changes pass through depths wise conv layers
+    m = module_masks.module
+    if m.in_channels == m.out_channels == m.groups:
         module_masks.output_mask = mask
-    # infer shape of parameters
-    weight_cmask = CoarseMask(num_dim=4)
-    weight_cmask.add_index_mask(dim=0, index=mask.mask_index[1])
-    bias_cmask = CoarseMask(num_dim=1)
-    bias_cmask.add_index_mask(dim=0, index=mask.mask_index[1])
-    module_masks.set_param_masks('weight', weight_cmask)
-    module_masks.set_param_masks('bias', bias_cmask)
-    # input shape is not changed
```

Comment:
output mask affects weight mask, why remove this part?

Reply:
Thanks, fixed.